### PR TITLE
HIVE-24652: If compactor worker times out, compaction is not cleared from queue

### PR DESCRIPTION
**Pull request by klcopp**
### Does this PR introduce _any_ user-facing change?

See HIVE-24652

### How was this patch tested?
Unit test

### CODE REVIEW

**Comment by klcopp** [comment]
Note: The compaction's txn will stay open if it is open when the Worker times out. But AcidHouseKeeperService should close it when the txn times out.

File: ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Worker.java [hunk 1]
```diff
@@ -102,11 +102,16 @@ public void run() {
     ExecutorService executor = getTimeoutHandlingExecutor();
     try {
       do {
-        Future<Boolean> singleRun = executor.submit(() -> findNextCompactionAndExecute(computeStats));
+        Ref<CompactionInfo> ci = new Ref<>(null);
+        Future<Boolean> singleRun = executor.submit(() -> findNextCompactionAndExecute(computeStats, ci));
```
Comment (pvary) [inline ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Worker.java:106]: Wouldn't it be more natural to have a complex return type, like below? Definitely not a final proposal just some food for thought.
```
private static class ExecutionResult {
   private Boolean singleRun;
   private CompactionInfo failedCompaction;
}
```

**Comment by pvargacl** [comment]
> Note: The compaction's txn will stay open if it is open when the Worker times out. But AcidHouseKeeperService should close it when the txn times out.

The txnId is saved in the compactionQueue when updateCompactorState is called, so you can abort it manually, if you want.

**Comment by klcopp** [comment]
Initiator#recoverFailedCompactions already takes care of this.

Thanks for taking a look and pointing this out @pvary !

**Pull request closed without merging**